multinet/api/views/workspace.py

```diff
@@ -25,6 +25,21 @@
 from .common import MultinetPagination
 
 
+def build_user_list(validated_data: OrderedDict) -> list:
+    """
+    Build a list of user objects from an ordered dictionary of user data.
+
+    Accepts an ordered dictionary containing a list of validated user data, e.g.
+    as a result of validating a PermissionsSerializer with request data.
+    Returns a list of user objects.
+    """
+    user_list = []
+    for valid_user in validated_data:
+        user_object = get_object_or_404(User, username=valid_user['username'])
+        user_list.append(user_object)
+    return user_list
```

Comment:
What's the rationale behind moving this function outside of `WorkspaceViewSet`? It's really just a helper function for `WorkspaceViewSet::put_workspace_permissions`. Do you see it having a use elsewhere? What if we indicated that it should be a private function with double underscores?

Reply:
`build_user_list` doesn't make use the `self` argument at all, so there's no functional reason it needs to reside inside of `WorkspaceViewSet`. I don't see it being used elsewhere, but I also don't see a problem with it living in `views/workspace.py`. It's still closely coupled, just not within `WorkspaceViewSet` itself. Essentially, I don't see a compelling reason to _keep it_ inside of `WorkspaceViewSet`.

Reply:
Is there a way to avoid making multiple DB requests in a loop here?